### PR TITLE
LPD-34097 Setting "noindex" or "nofollow" in robots does not remove page from sitemap

### PR DESCRIPTION
**Pull request by jonathanmccann**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-34097

# How am I fixing it?

In https://liferay.atlassian.net/browse/LPD-9969 the field for `Sitemap Include` was disabled whenever the `robots` contained `nofollow` or `noindex`. However, disabled fields are not submitted within forms so in the backend we only received a `null` value - https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-seo-web/src/main/resources/META-INF/resources/js/seo/toggleSitemapFields.js#L54-L56

With these changes it checks to see if the robots contain either of those phrases and explicitly sets the `sitemap-include` to `0` (`false`).

# How can you verify that it works?

In `modules/apps/layout/layout-seo-test` run `gw testIntegration --tests *.EditSEOMVCActionCommandTest`

Manual steps can be found on https://liferay.atlassian.net/browse/LPD-34097

### CODE REVIEW

**Comment by jonathanmccann** [comment]
ci:test:relevant

**Comment by jonathanmccann** [comment]
ci:test:relevant

**Comment by jonathanmccann** [comment]
@david-truong @mrjohn1911 Would either of you mind reviewing these changes?

File: modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/portlet/action/EditSEOMVCActionCommand.java [hunk 1]
```diff
@@ -98,6 +101,20 @@ protected void doProcessAction(
...
+
+			if (Validator.isNotNull(value) &&
+				(StringUtil.containsIgnoreCase(
+					value, "nofollow", StringPool.BLANK) ||
+				 StringUtil.containsIgnoreCase(
+					 value, "noindex", StringPool.BLANK))) {
+
+				formTypeSettingsUnicodeProperties.setProperty(
+					LayoutTypePortletConstants.SITEMAP_INCLUDE, "0");
+			}
```
Comment (david-truong) [inline modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/portlet/action/EditSEOMVCActionCommand.java:115]: Can you break out of the loop here also?

Comment (jonathanmccann) [reply modules/apps/layout/layout-seo-web/src/main/java/com/liferay/layout/seo/web/internal/portlet/action/EditSEOMVCActionCommand.java:117]: Yes, good catch, thank you.

**Comment by jonathanmccann** [comment]
ci:test:sf

**Comment by jonathanmccann** [comment]
@david-truong Let me know if there's anything else needing to be changed or if this can be forwarded. Thanks!

**Comment by david-truong** [comment]
Yes you can forward

**Comment by jonathanmccann** [comment]
ci:forward

**Pull request closed without merging**